### ISSUE TITLE
noproxy: true type mismatch

### ISSUE THREAD

**Issue by unrealsolver** [avkonst/hookstate#413]
When using .get with `{noproxy: true}` the return type stated as `Immutable<S>`, however should be just `S`.
```ts
get(options?: { noproxy?: boolean, stealth?: boolean }): Immutable<S>;
```

Due to the implementation details, it would be hard to write a conditional type here. Options are:
- Create a `.getBare` or `.getSource` method or something similar, which would have noproxy behavior
- Change `options?` to positional args with conditional types

**Comment by avkonst** [comment]
Well, technically it is immutable, because changing it would almost certainly be a bug...

**Comment by unrealsolver** [comment]
And totally incompatible.  So I'd like to have a solution better then explicit cast with `as`. Not sure what would be the best. Maybe replacing `Immutable` with implicit agreement or something like that

**Comment by avkonst** [comment]
@speigg as your team is the biggest user of the hookstate, could you please suggest what compromise we should have here?

**Comment by avkonst** [comment]
There is also this similar: https://github.com/avkonst/hookstate/issues/421

**Comment by avkonst** [comment]
The only alternative is to go back and drop Immutable everywhere... not sure

**Comment by unrealsolver** [comment]
Right. So `Immutable*` is just nominally telling that direct assignments to the state are not allowed. Which could also be achieved through these:
* Formal convention. I think `useState` is just telling to use `setter` function instead of direct assignment in the docs and no checks if someone would try to set state with direct assignment.
* use `Proxy` for the `NODE_ENV=development` which would just print scary error telling the user not to do direct assignments.
